### PR TITLE
implement retries

### PR DESCRIPTION
**Pull request by aviau**
WIP for now.

I think this does not play well with requests-mock, so we can't test it really.

### CODE REVIEW

File: flareio/api_client.py [hunk 1]
```diff
@@ -24,7 +26,31 @@ def __init__(
...
+    def _create_session() -> requests.Session:
+        session = requests.Session()
+        retries = Retry(
+            total=5,
+            backoff_factor=1,
+            status_forcelist=[429, 502, 503, 504],
+            allowed_methods={"GET", "POST"},
+            backoff_max=15,
+        )
+        session.mount(
```
Comment (isra17) [inline flareio/api_client.py:41]: Why mount for `http://` ?

Comment (aviau) [reply flareio/api_client.py:41]: testing stuff...

Comment (aviau) [reply flareio/api_client.py:41]: We can't really test it. So I think i'll remove the unit test and just merge and hope for the best

**Comment by aviau** [comment]
WDYT @isra17 ?

**Comment by isra17** [comment]
I don't think I would enable retry by default. Actually, the more I think about it, the more I would rather just document how to do it through custom session and not build this in the client sdk.

**Comment by aviau** [comment]
Hmm, I don't mind documenting custom sessions, but I think this might fix problems for most users at little cost.

**Review by isra17 — APPROVED** [review]

**Pull request merged**